### PR TITLE
feat: added compareKey(target)

### PR DESCRIPTION
**Pull request by swittk**
Added `compareKey` to the iterator so it's possible to compare the current key of the iterator to a string/ArrayBuffer using the underlying `Compare()` method.

I'm trying to implement a `abstract-level` compliant wrapper around this library and found that this should help with the implementation of a `gt`, `gte`, `lt`, `lte` requirements for the iterator.

### CODE REVIEW

**Comment by savv** [comment]
Hi @swittk ! Thanks a lot for this PR. Sorry I didn't see it earlier. Will review ASAP and get back to you

File: example/ios/LeveldbExample.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>IDEDidComputeMac32BitWarning</key>
```
Comment (savv) [inline example/ios/LeveldbExample.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist:5]: what is this file? If it's specific to your ID or setup, could we remove it?

File: example/ios/LeveldbExample.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
```
Comment (savv) [inline example/ios/LeveldbExample.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings:3]: same comment re: being setup specific

**Review by savv — APPROVED** [review]

File: example/ios/LeveldbExample.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist (hunk 1 above)
Comment (swittk) [reply example/ios/LeveldbExample.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist:5]: Hi. Seems to be a file generated by all Xcode with versions > 9.3 (which was a while ago; right now is version 13+).
https://stackoverflow.com/a/50368149/4469172

Comment (swittk) [reply example/ios/LeveldbExample.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist:5]: But I think removing xcshareddata should be good.

Comment (swittk) [reply example/ios/LeveldbExample.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist:5]: Ah. according to here xcshareddata should be committed though
https://stackoverflow.com/a/53039267/4469172
`Xcode 9.3 adds a new IDEWorkspaceChecks.plist file to a workspace's shared data, to store the state of necessary workspace checks. Committing this file to source control will prevent unnecessary rerunning of those checks for each user opening the workspace. (37293167)`

So... I guess we should keep them.

**Review by savv — APPROVED** [review]
thanks for checking and for adding this to our lib!

**Comment by savv** [comment]
ah. I guess I have to merge it myself.

**Comment by shamilovtim** [comment]
@swittk how did your abstract level work go? I was just doing the same thing and seeing if you published anything for that

**Comment by swittk** [comment]
I did publish it a while back; just came back to this after searching my previous commit history.
https://github.com/swittk/react-native-leveldb-level-adapter
Just adapted it to use the latest version of this lib because I needed something as a more tried and true backend for PouchDB on react native.

**Pull request merged**